onnxruntime/contrib_ops/rocm/bert/fast_gelu_impl.cu

```diff
@@ -60,88 +60,59 @@ __global__ void FastGeluKernel(const T a, const T b, const T c, const T oneT, co
   }
 }
 
-template <unsigned TPB>
-__global__ void FastGeluKernel2(const half2 a, const half2 b, const half2 c, const half2 one2, const half2 two2,
-                                int input_length, int bias_length, const half2* input, const half2* bias,
-                                half2* output) {
-  const int idx = blockIdx.x * TPB + threadIdx.x;
-
+template <typename T, unsigned TPB, int ILP>
+__global__ void FastGeluBiasKernelVec(const T a, const T b, const T c, const T oneT, const T twoT,
```

Comment:
Can we have the tail handling in the code? This will allow us to delete the original non-vectorized code (so that a single version of kernel is maintained), as well as to simplify the branching logic for kernel launch.